feature/gribi/otg_tests/supervisor_failure_test/supervisor_failure_test.go

```diff
@@ -0,0 +1,382 @@
+	} else {
+		t.Logf("Found primary.LastSwitchoverTime(): %v", gnmi.Get(t, dut, primary.LastSwitchoverTime().State()))
+	}
+
+	if !gnmi.Lookup(t, dut, primary.LastSwitchoverReason().State()).IsPresent() {
+		t.Errorf("primary.LastSwitchoverReason().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastSwitchoverReason := gnmi.Get(t, dut, primary.LastSwitchoverReason().State())
+		t.Logf("Found lastSwitchoverReason.GetDetails(): %v", lastSwitchoverReason.GetDetails())
+		t.Logf("Found lastSwitchoverReason.GetTrigger().String(): %v", lastSwitchoverReason.GetTrigger().String())
+	}
+	if gnmi.Get(t, dut, primary.LastSwitchoverReason().State()).GetTrigger() != switchTrigger {
+		t.Errorf("primary.GetLastSwitchoverReason().GetTrigger(): got %s, want SYSTEM_INITIATED.",
+			gnmi.Get(t, dut, primary.LastSwitchoverReason().State()).GetTrigger().String())
+	}
+
+	if !gnmi.Lookup(t, dut, primary.LastRebootTime().State()).IsPresent() {
+		t.Errorf("primary.LastRebootTime.().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastrebootTime := gnmi.Get(t, dut, primary.LastRebootTime().State())
+		t.Logf("Found lastRebootTime.GetDetails(): %v", lastrebootTime)
+	}
+	if !gnmi.Lookup(t, dut, primary.LastRebootReason().State()).IsPresent() {
+		t.Errorf("primary.LastRebootReason.().Lookup(t).IsPresent(): got false, want true")
+	} else {
+		lastrebootReason := gnmi.Get(t, dut, primary.LastRebootReason().State())
+		t.Logf("Found lastRebootReason.GetDetails(): %v", lastrebootReason)
+	}
+}
+
+func switchoverReady(t *testing.T, dut *ondatra.DUTDevice, controller string) bool {
+	switchoverReady := gnmi.OC().Component(controller).SwitchoverReady()
+	_, ok := gnmi.Watch(t, dut, switchoverReady.State(), 30*time.Minute, func(val *ygnmi.Value[bool]) bool {
+		ready, present := val.Val()
+		return present && ready
+	}).Await(t)
+	return ok
+}
+
+func TestSupFailure(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+
+	// Configure the DUT
+	configureDUT(t, dut)
+
+	// Configure the ATE
+	ate := ondatra.ATE(t, "ate")
+	top := configureATE(t, ate)
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
+
+	// Configure the gRIBI client clientA
+	clientA := gribi.Client{
+		DUT:         dut,
+		FIBACK:      false,
+		Persistence: true,
+	}
+	defer clientA.Close(t)
+
+	// Flush all entries after test.
+	defer clientA.FlushAll(t)
+
+	if err := clientA.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection can not be established")
+	}
+	clientA.BecomeLeader(t)
+
+	// Flush all entries before test.
+	clientA.FlushAll(t)
+
+	args := &testArgs{
+		ctx:     ctx,
+		clientA: &clientA,
+		dut:     dut,
+		ate:     ate,
+		top:     top,
+	}
+	// Program a route and ensure AFT telemetry returns FIB_PROGRAMMED
+	routeInstall(ctx, t, args)
+	// Verify that static route(203.0.113.0/24) to ATE port-2 is preferred by the traffic.`
+	sendTraffic(t, args.ate)
+	time.Sleep(15 * time.Second)
+	otgutils.LogFlowMetrics(t, ate.OTG(), top)
```

Comment:
The send/stopTraffic functions are not encapsulating much, it will be more clear to just inline their functionality here.

Reply:
Done